pkg/util/compression/impl-noop/no_strategy.go

```diff
@@ -9,23 +9,15 @@ package noopimpl
 import (
 	"bytes"
 
-	compression "github.com/DataDog/datadog-agent/comp/serializer/compression/def"
+	"github.com/DataDog/datadog-agent/pkg/util/compression"
 )
 
-// Provides contains the compression component
-type Provides struct {
-	Comp compression.Component
-}
-
 // NoopStrategy is the strategy for when serializer_compressor_kind is neither zlib nor zstd
-type NoopStrategy struct {
```

Comment:
I can't leave a comment on the affected line - but I think `ContentEncoding` needs to return `identity` in order to be consistent with the now removed `identityContentType` used by the log agent. 

(TBH i'm not sure if this would cause any problems as is, but it doesn't hurt to maintain consistency)